server/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -102,25 +101,25 @@ StoredScriptSource getScriptFromClusterState(String id) {
     // simply by multiplying by five, so even setting it to one, requires five compilations to break
     public void testCompilationCircuitBreaking() throws Exception {
         buildScriptService(Settings.EMPTY);
-        scriptService.setMaxCompilationRate(Tuple.tuple(1, TimeValue.timeValueMinutes(1)));
-        scriptService.checkCompilationLimit(); // should pass
-        expectThrows(CircuitBreakingException.class, () -> scriptService.checkCompilationLimit());
-        scriptService.setMaxCompilationRate(Tuple.tuple(2, TimeValue.timeValueMinutes(1)));
-        scriptService.checkCompilationLimit(); // should pass
-        scriptService.checkCompilationLimit(); // should pass
-        expectThrows(CircuitBreakingException.class, () -> scriptService.checkCompilationLimit());
+        scriptService.compiler.setMaxCompilationRate(Tuple.tuple(1, TimeValue.timeValueMinutes(1)));
```

Comment:
This seems like a series of unit tests on ContextCompiler, not a test of ScriptService.  Can we add a ContextCompilerTests, and move this test there?

Reply:
Moved to `ScriptCacheTests`.